### PR TITLE
Bugfix/add coverage to sonarcloud

### PR DESCRIPTION
**Pull request by mahdiatallah**
### PR Description

This PR updates our SonarCloud configuration to recognize new code properly. This update also improves our overall code quality metrics by integrating test coverage as a component of the SonarCloud quality gate as well as other metrics such as duplication. The previous configuration did not effectively track new code, so this adjustment aligns our setup with best practices for monitoring recent changes.

### Key Points
- **Test Prerequisite**: SonarCloud’s analysis now requires all tests to pass. Currently, some issues with our Jest setup are preventing the quality gate from passing, but this will resolve once those test issues are fixed.
- **Coverage Focus**: Coverage analysis is restricted to new code introduced in the PR, providing targeted insights into recent changes.

### Testing Instructions
To validate this configuration:
1. **Branch Creation**: Create a new branch from this PR.
2. **Adjust Tests**: Remove existing tests with import issues to avoid failures.
3. **Add New Code**: Introduce new code, as SonarCloud focuses coverage stats on recent additions.
4. **Pass Tests**: Write and verify passing tests for the new code.
5. **Create PR and Verify**: Submit a PR with these changes and review the quality gate metrics by following the link in the PR.

### Documentation
Refer to the [SonarCloud coverage documentation](https://docs.sonarsource.com/sonarcloud/enriching/test-coverage/overview/) for more details on setup and options for enriching coverage.

*P.S.* Helpful tip for anyone working on CI/CD: Files created in a workflow are cleaned up at the end of each job, which is why I’ve configured the tests to run directly in the SonarCloud YAML file. To ensure coverage files are retained for analysis, tests and analysis need to be in the same workflow. Previously, coverage files were generated in the `node.js` workflow, but they weren’t accessible to the SonarCloud workflow due to cleanup after each job. This is why running tests within the SonarCloud workflow is necessary.

### CODE REVIEW

**Review by Rcc999 — COMMENTED** [review]
SonarCloud Configuration Improvements:

Good job on integrating SonarCloud more effectively by focusing on new code and coverage metrics. This will help in maintaining code quality incrementally without overwhelming the team with legacy code issues.


Node.js Workflow Removal:

If the node.js.yml workflow isn’t providing value anymore or conflicts with the current CI/CD structure, you should remove it for better maintenance. Alternatively, if it’s deprecated but still in use for edge cases, clearly document why it exists to avoid confusion.

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sonar.c.file.suffixes=-
+sonar.cpp.file.suffixes=-
+sonar.objc.file.suffixes=-
```
[the remark this answers is not in the document]
Comment (mahdiatallah) [reply sonar-project.properties:3]: Interestingly, the project includes some C/C++/Objective C files, which require separate configuration in SonarCloud. These files aren’t ones we actively write—they’re part of imported packages—so this update excludes them from analysis. This adjustment was suggested by SonarCloud itself, which flagged these files during analysis and was unable to proceed.

**Comment by mahdiatallah** [comment]
I'm glad you like the PR!

Good point on `node.js.yml`. Since our testing is already handled in SonarCloud, there’s indeed no need for `node.js.yml`, so I'll go ahead and remove it.

This also made me realize I should add a note explaining why I run the tests directly in the SonarCloud YAML file, so that anyone handling CI/CD in the future can understand the setup. Specifically, tests and analysis need to be run in the same workflow because each worker creates files and then cleans up afterward. Although we were generating coverage files in `node.js.yml`, they couldn’t be accessed by `sonarcloud.yml` since they were deleted after the job completed. This is why running the tests within the SonarCloud workflow is necessary.

I'll add this clarification in the comments for future reference.

**Comment by mahdiatallah** [comment]
I’ve also removed the `coverage` folder, as it shouldn’t be pushed to the remote repository.

**Review by Rcc999 — APPROVED** [review]

**Pull request merged**